typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -86,4 +186,610 @@ export class EvmIsmModule extends HyperlaneModule<
+        try {
+          return domainRoutingIsmFactory.interface.parseLog(log);
+        } catch (e) {
+          return undefined;
+        }
+      })
+      .filter(
+        (log): log is ethers.utils.LogDescription =>
+          !!log && log.name === 'ModuleDeployed',
+      );
+
+    if (dispatchLogs.length === 0) {
+      throw new Error('No ModuleDeployed event found');
+    }
+
+    const moduleAddress = dispatchLogs[0].args['module'];
+    return DomainRoutingIsm__factory.connect(moduleAddress, signer);
+  }
+
+  protected static async deployAggregationIsm(params: {
+    destination: ChainName;
+    config: AggregationIsmConfig;
+    origin?: ChainName;
+    mailbox?: Address;
+    logger: Logger;
+    multiProvider: MultiProvider;
+    factories: HyperlaneAddresses<ProxyFactoryFactories>;
+    deployer: HyperlaneDeployer<any, any>;
+  }): Promise<IAggregationIsm> {
+    const {
+      destination,
+      config,
+      origin,
+      mailbox,
+      logger,
+      multiProvider,
+      factories,
+      deployer,
+    } = params;
+
+    const signer = multiProvider.getSigner(destination);
+
+    const addresses: Address[] = [];
+    for (const module of config.modules) {
+      const submodule = await EvmIsmModule.deploy({
+        destination,
+        config: module,
+        origin,
+        mailbox,
+        multiProvider,
+        factories,
+        deployer,
+      });
+      addresses.push(submodule.address);
+    }
+
+    const factory = StaticAggregationIsmFactory__factory.connect(
+      factories.staticAggregationIsmFactory,
+      signer,
+    );
+
+    const address = await EvmIsmModule.deployStaticAddressSet({
+      chain: destination,
+      factory,
+      values: addresses,
+      logger: logger,
+      threshold: config.threshold,
+      multiProvider,
+    });
+    return IAggregationIsm__factory.connect(address, signer);
+  }
+
+  protected static async deployStaticAddressSet(params: {
+    chain: ChainName;
+    factory: StaticThresholdAddressSetFactory | StaticAddressSetFactory;
+    values: Address[];
+    logger: Logger;
+    threshold?: number;
+    multiProvider: MultiProvider;
+  }): Promise<Address> {
+    const {
+      chain,
+      factory,
+      values,
+      logger,
+      threshold = values.length,
+      multiProvider,
+    } = params;
+    const sorted = [...values].sort();
+
+    const address = await factory['getAddress(address[],uint8)'](
+      sorted,
+      threshold,
+    );
+    const code = await multiProvider.getProvider(chain).getCode(address);
+    if (code === '0x') {
+      logger.debug(
+        `Deploying new ${threshold} of ${values.length} address set to ${chain}`,
+      );
+      const overrides = multiProvider.getTransactionOverrides(chain);
+      const hash = await factory['deploy(address[],uint8)'](
+        sorted,
+        threshold,
+        overrides,
+      );
+      await multiProvider.handleTx(chain, hash);
+      // TODO: add proxy verification artifact?
```

Comment:
could do something like this but not sure how to get the constructor arguments - if you know a quick way then I'll do it but otherwise don't want to spend too much time on this
<img width="546" alt="image" src="https://github.com/hyperlane-xyz/hyperlane-monorepo/assets/10051819/5c89dea1-1f7a-4dce-8bce-b79c99eb96b9">

Reply:
could also include verifier in the main class, and have the non-static callers of deployStaticAddressSet call the verifyContract method